tools/build_info.hh

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+#pragma once
+
+#include "build_mode.hh"
+
+namespace tools::build_info {
+enum class build_type { debug = 0, release, dev, sanitize, coverage, unknown };
+
+constexpr build_type get_build_type() {
+#ifdef SCYLLA_BUILD_MODE_DEBUG
+    return build_type::debug;
+#endif
+#ifdef SCYLLA_BUILD_MODE_RELEASE
+    return build_type::release;
+#endif
+#ifdef SCYLLA_BUILD_MODE_DEV
+    return build_type::dev;
+#endif
+#ifdef SCYLLA_BUILD_MODE_SANITIZE
+    return build_type::sanitize;
+#endif
+#ifdef SCYLLA_BUILD_MODE_COVERAGE
+    return build_type::coverage;
+#endif
+    return build_type::unknown;
```

Comment:
Should we fail compilation if a mode is unknown?

Reply:
Yep, probably better than just silently fail.

Reply:
Changed it to fail in the new version.